[Hack Week] ApiFaker: display a hint on the main app UI when the ApiFaker is enabled

hichamboushaba opened the pull request:
### Description
As part of my Hack Week, I'm bringing back the ApiFaker project paqN3M-SI-p2, and I'm splitting the feature into multiple PRs.

This PR adds a UI bottom banner to the main app when the ApiFaker is enabled, and the banner allows an easy path to disabling the ApiFaker.

I added `unit-tests-exemption` label because the class ApiFakerUiHelper has UI logic and can't be unit tested.
 
### Steps to reproduce
1. Use a debug build.
2. Open the app settings.
3. Tap on Developer Options.
4. Tap on ApiFaker.
5. Add some fake endpoints definitions.
6. Enable the ApiFaker.

### Testing information
- Confirm that when enabled, a bottom banner is shown on all app Activities (Login, MainActivity, Settings)
- Confirm that tapping on the banner shows a dialog that allows disabling the ApiFaker.

### The tests that have been performed
^

### Images/gif
<img src=https://github.com/user-attachments/assets/f280cb93-24ae-4ac6-b342-ddaa05d900d2 width=360/>
<img src=https://github.com/user-attachments/assets/9e85129b-267e-4f5e-8d8a-62aefd6eba8e width=360/>

</br></br>

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

hichamboushaba committed:
Display a hint when ApiFaker is enabled

hichamboushaba committed:
Add ability to disable the ApiFaker from the UI hint

hichamboushaba committed:
Add query parameters to the DB model

hichamboushaba committed:
Filter requests using query parameters

hichamboushaba committed:
Add a way to add query parameters in the UI

hichamboushaba committed:
Some fixes

hichamboushaba committed:
Log a warning message if multiple endpoints are matched

hichamboushaba committed:
Bump db version

hichamboushaba committed:
Handle query parameters extraction for Jetpack tunnel requests

hichamboushaba committed:
Set text color as white in the hint

hichamboushaba committed:
Fix detekt issues

hichamboushaba committed:
Fix detekt issues

dangermattic commented:
<!--
  0 Errors
  3 Warnings: This PR is larger than 300 lin...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>ApiFakerUiHelper</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>QueryParameterConverter</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

hichamboushaba committed:
Add more tests

hafizrahman reviewed (approved):
Followed the test instruction and banner appeared correctly and dialog also worked 👍🏼 

Code looks good too.

hichamboushaba committed:
Improve logic of warning log

This makes sure the log is not printed when the matched endpoints use different query parameters.

hichamboushaba committed:
Merge pull request #13083 from woocommerce/task/api-faker-5

[Hack Week] ApiFaker: match endpoints using query parameters
